examples_apps/run_gate_synthesis.py

```diff
@@ -0,0 +1,470 @@
+
+
+######################################################################
+# Results and visualisation
+# -------------------------
+#
+# Plotting the cost vs. optimization step:
+#
+
+from matplotlib import pyplot as plt
+# %matplotlib inline
+plt.rcParams['font.family'] = 'serif'
+plt.rcParams['font.sans-serif'] = ['Computer Modern Roman']
+plt.style.use('default')
+
+plt.plot(cost_progress)
+plt.ylabel('Cost')
+plt.xlabel('Step')
+plt.show()
+
+
+######################################################################
+# We can use matrix plots to plot the real and imaginary components of the
+# target unitary :math:`V` and learnt unitary :math:`U`.
+#
+
+learnt_unitary = ket_val.numpy().T[:gate_cutoff, :gate_cutoff]
+target_unitary = target_unitary[:gate_cutoff, :gate_cutoff]
+
+fig, ax = plt.subplots(1, 4, figsize=(7, 4))
+ax[0].matshow(target_unitary.real, cmap=plt.get_cmap('Reds'))
+ax[1].matshow(target_unitary.imag, cmap=plt.get_cmap('Greens'))
+ax[2].matshow(learnt_unitary.real, cmap=plt.get_cmap('Reds'))
+ax[3].matshow(learnt_unitary.imag, cmap=plt.get_cmap('Greens'))
+
+ax[0].set_xlabel(r'$\mathrm{Re}(V)$')
+ax[1].set_xlabel(r'$\mathrm{Im}(V)$')
+ax[2].set_xlabel(r'$\mathrm{Re}(U)$')
+ax[3].set_xlabel(r'$\mathrm{Im}(U)$')
+fig.show()
+
+
+######################################################################
+# Process fidelity
+# ----------------
+#
+# The process fidelity between the two unitaries is defined by
+#
+# .. math:: F_e  = \left| \left\langle \Psi(V) \mid \Psi(U)\right\rangle\right|^2
+#
+# where:
+#
+# -  :math:`\left|\Psi(V)\right\rangle` is the action of :math:`V` on one
+#    half of a maximally entangled state :math:`\left|\phi\right\rangle`:
+#
+# .. math:: \left|\Psi(V)\right\rangle = (I\otimes V)\left|\phi\right\rangle,
+#
+# -  :math:`V` is the target unitary,
+# -  :math:`U` the learnt unitary.
+#
+
+I = np.identity(gate_cutoff)
+phi = I.flatten()/np.sqrt(gate_cutoff)
+psiV = np.kron(I, target_unitary) @ phi
+psiU = np.kron(I, learnt_unitary) @ phi
+
+
+
+######################################################################
+# Therefore, after 200 repetitions, the learnt unitary synthesized via a
+# variational quantum circuit has the following process fidelity to the target
+# unitary:
+#
+
+print(np.abs(np.vdot(psiV, psiU))**2)
+
+
+######################################################################
+# Circuit parameters
+# ------------------
+#
+# We can also query the optimal variational circuit parameters
+# :math:`\vec{\theta}` that resulted in the learnt unitary. For example,
+# to determine the maximum squeezing magnitude in the variational quantum
+# circuit:
+#
+
+print(np.max(np.abs(weights[:, 0])))
+
+######################################################################
+# Further results
+# ---------------
+#
+# Even more refined results can be obtained by increasing the number of
+# repetitions (``reps``) after downloading the tutorial!
```

Comment:
Nice! Could even suggest playing around with the number of layers and the gate cutoff

Reply:
Added!